test/test_upload_processor.py

```diff
@@ -0,0 +1,157 @@
+""" This file tests the functions defined in upload_processor """
+
+import random
+import string
+import unittest
+import subprocess
+import os
+from typing import List
+import hailtop.batch as hb
+from joint_calling.upload_processor import batch_move_files
+
+
+def validate_move(upload_prefix: str, main_prefix: str, sample: str) -> bool:
+    """Checks that a given file exists at the location in
+    the main bucket and no longer exists in the upload bucket.
+    Returns True if this is the case and False otherwise.
+    """
+
+    main_path = os.path.join('gs://', main_prefix, sample)
+    upload_path = os.path.join('gs://', upload_prefix, sample)
+
+    exists_main = subprocess.run(['gsutil', '-q', 'stat', main_path], check=False)
+    exists_upload = subprocess.run(['gsutil', '-q', 'stat', upload_path], check=False)
+
+    # Exists at destination and not at source
+    return exists_upload.returncode != 0 and exists_main.returncode == 0
+
+
+def upload_files(files: List[str], upload_prefix: str):
+    """A function to mimic file upload. Takes a list of
+    file names, creates these files, then moves them into
+    the upload bucket used for further testing."""
+
+    for f in files:
+        subprocess.run(['touch', f], check=True)
+        full_path = os.path.join('gs://', upload_prefix, f)
+        subprocess.run(['gsutil', 'mv', f, full_path], check=True)
+
+
+class TestUploadProcessor(unittest.TestCase):
+    """Test cases for the upload processor"""
+
+    def setUp(self):
+        """Initialises standard variables for testing"""
+
+        # Create random string of digits to label current test folder
+        random_digits = ''.join(random.choice(string.digits) for i in range(5))
+        test_folder = 'test' + random_digits
```

Comment:
If it's not too much work, adding the current date in the folder name can be handy, too. This way it's easy to see whether a folder has been created ages ago and can safely be deleted.

Reply:
Good idea :) Have added it!